src/components/cc-addon-backups/cc-addon-backups.smart.js

```diff
@@ -0,0 +1,170 @@
+import { sendToApi } from '../../lib/send-to-api.js';
+import { defineSmartComponent } from '../../lib/smart/define-smart-component.js';
+import '../cc-smart-container/cc-smart-container.js';
+import './cc-addon-backups.js';
+// @ts-expect-error FIXME: remove when clever-client exports types
+import { get as getAddon } from '@clevercloud/client/esm/api/v2/addon.js';
+// @ts-expect-error FIXME: remove when clever-client exports types
+import { getBackups } from '@clevercloud/client/esm/api/v2/backups.js';
+// @ts-expect-error FIXME: remove when clever-client exports types
+import { getAddon as getAddonFromProvider } from '@clevercloud/client/esm/api/v4/addon-providers.js';
+
+/**
+ * @typedef {import('../common.types.js').Addon} Addon
+ * @typedef {import('../common.types.js').AddonProvider} AddonProvider
+ * @typedef {import('./cc-addon-backups.js').CcAddonBackups} CcAddonBackups
+ * @typedef {import('../../lib/send-to-api.types.js').ApiConfig} ApiConfig
+ * @typedef {import('../../lib/smart/smart-component.types.js').OnContextUpdateArgs<CcAddonBackups>} OnContextUpdateArgs
+ */
+defineSmartComponent({
+  selector: 'cc-addon-backups',
+  params: {
+    apiConfig: { type: Object },
+    ownerId: { type: String },
+    addonId: { type: String },
+  },
+  /**
+   * @param {OnContextUpdateArgs} args
+   */
+  async onContextUpdate({ context, updateComponent, signal }) {
+    updateComponent('state', { type: 'loading' });
+    const { apiConfig, ownerId, addonId } = context;
+
+    const api = getApi(apiConfig, signal);
+    const addon = await api.fetchAddon({ ownerId, addonId });
+    const addonDetails = await api.fetchAddonDetails({ addon, addonId });
+    const rawBackups = await api.fetchBackups({ addon, ownerId, addonDetails });
+    const backups = rawBackups.sort(
+      (
+        /** @type {{ createdAt: string | number | Date; }} */ a,
+        /** @type {{ createdAt: string | number | Date; }} */ b,
+      ) => new Date(b.createdAt).getTime() - new Date(a.createdAt).getTime(),
+    );
+
+    updateComponent('state', {
+      type: 'loaded',
+      providerId: api.getProviderId({ addon, addonDetails }),
```

Comment:
nitpick: feel free to ignore that, but this confused me at first because I thought `getProviderId` was an API call, I'd probably move the `getProviderId` outside of the api object